src/main/java/org/apache/accumulo/proxy/ProxyServer.java

```diff
@@ -129,8 +125,10 @@ public class ProxyServer implements AccumuloProxy.Iface {
   public static final Logger logger = LoggerFactory.getLogger(ProxyServer.class);
   public static final String RPC_ACCUMULO_PRINCIPAL_MISMATCH_MSG = "RPC"
       + " principal did not match requested Accumulo principal";
-  @SuppressWarnings("deprecation")
-  protected org.apache.accumulo.core.client.Instance instance;
+
+  protected AccumuloClient client;
+
+  private final String secret;
```

Comment:
This is one of the main changes in this PR. The addition of this shared secret allows for the client to authenticate to the proxy service without telling the proxy service how to authenticate to accumulo.